SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/DBSPCompiler.java

```diff
@@ -94,11 +103,23 @@ enum InputSource {
 
     public DBSPCompiler(CompilerOptions options) {
         this.options = options;
+        this.mapper = new ObjectMapper();
         this.frontend = new CalciteCompiler(options);
         this.midend = new CalciteToDBSPCompiler(this.frontend, true, options, this);
         this.messages = new CompilerMessages(this);
         this.sources = new SourceFileContents();
         this.circuit = null;
+        if (options.ioOptions.emitJsonSchema != null) {
+            this.inputs = this.mapper.createArrayNode();
+            this.outputs = this.mapper.createArrayNode();
+            this.ios = this.mapper.createObjectNode();
+            this.ios.set("inputs", this.inputs);
```

Comment:
Ah, I see now that this is where you handle inputs and outputs separately. Got it.